packages/aws-cdk-lib/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -265,6 +265,16 @@ export interface BucketDeploymentProps {
    * @default - `x-amz-content-sha256` will not be computed
    */
   readonly signContent?: boolean;
+
+  /**
+   * If set to false, the custom resource will not send back the SourceObjectKeys.
+   * This is useful when you are facing the error `Response object is too long`
+   *
+   * See https://github.com/aws/aws-cdk/issues/28579
+   *
+   * @default true
```

Comment:
Might be missing it, but is there somewhere in the code we are defaulting this value to true? The closest thing I can find is `../product-stack-synthesizer.ts`, but that sets it to `false`.

Reply:
its value is set to true here https://github.com/aws/aws-cdk/blob/e610d056ec55db0f47d2eb09fff86c25806a0898/packages/@aws-cdk/custom-resource-handlers/lib/aws-s3-deployment/bucket-deployment-handler/index.py#L67 .. but you have a point, we should set it to true as well in the construct

Reply:
updated